### ISSUE TITLE
Qute: implement widening primitive conversion in templates

### ISSUE THREAD

**Issue by mkouba** [quarkusio/quarkus#40097]
### Description

This issue is a follow-up of https://github.com/quarkusio/quarkus/issues/40072.

**Comment by neon-dev** [comment]
Would this also fix checks like this one?
```smarty
{@long number=0L}
{number} is {#if number == 0}0{#else}not 0{/if}
```

**Comment by mkouba** [comment]
> Would this also fix checks like this one?
> 
> {@long number=0L}
> {number} is {#if number == 0}0{#else}not 0{/if}

I don't think so. Parameter declarations (`{@long number=0L}`) are only considered during build validation whereas the problem mentioned in #40072 occurs at runtime, when the template is rendered - an `int` literal is used in the template but the method accepts `long`.

**Comment by neon-dev** [comment]
Ok thanks. I created #44610, as I couldn't find an existing issue.
